js/components/immutableComponent.js

```diff
@@ -7,7 +7,7 @@ const Immutable = require('immutable')
 
 class ImmutableComponent extends React.Component {
   shouldComponentUpdate (nextProps, nextState) {
-    return Object.keys(nextProps).some(prop => !Immutable.is(nextState, this.props[prop]))
+    return !Immutable.is(Immutable.fromJS(this.props), Immutable.fromJS(nextProps))
```

Comment:
Please double verify that Immutable.formJS will not cause any slowdown if you already have an immutable object. I bet it is fine, but please double check.  Just because the performance of the entire app relies on this check.